engine/src/main/java/com/datatorrent/stram/engine/GenericNode.java

```diff
@@ -608,9 +610,9 @@ else if (tracker.ports[trackerIndex] == null) {
           if (need2sleep) {
             if (handleIdleTime && insideWindow) {
               ((IdleTimeHandler) operator).handleIdleTime();
-            }
-            else {
+            } else {
               Thread.sleep(spinMillis);
+              spinMillis = Math.min(maxSpinMillis, spinMillis + 1);
```

Comment:
I also thought about such approach with a variation of providing a configuration setting for a number of iterations instead of a timeout as getting current time on each iteration may be way too costly.